src/pyseekdb/utils/embedding_functions/text2vec_embedding_function.py

```
@@ -0,0 +1,139 @@
+    def __init__(
+        self,
+        model_name: str = "shibing624/text2vec-base-chinese",
+        device: str = "cpu",
+        normalize_embeddings: bool = False,
+        **kwargs: Any,
+    ):
+        """Initialize Text2VecEmbeddingFunction."""
+        # Validate kwargs - only allow primitive types
+        for key, value in kwargs.items():
+            if not isinstance(value, (str, int, float, bool, list, dict, tuple)):
+                raise TypeError(f"Keyword argument {key} is not a primitive type")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)